### PR TITLE
build: auto update

### PR DESCRIPTION
**Pull request by mrfelton**
This builds on https://github.com/LN-Zap/zap-desktop/pull/599 and adds support for auto updates using electron-updater.

I've tested this locally using the following steps:

## Install minio
Minio is a private cloud storage solution with an API that is compatible with AWS. We use it to host update files that we will make available to electron-updater.
```
brew install minio/stable/minio
```
See https://www.minio.io/ for info about Mino.

## Create minio data dir and test bucket
```
mkdir -p ~/minio-data/test-update
```

## Start minio server
```
minio server ~/minio-data
```

## Make bucket public
Open up the minio web UI and add a READ policy on the test-update bucket
screenshot 2018-07-31 17 29 18
screenshot 2018-07-31 17 29 31

## Build and package the app to the minio bucket
Replace `[YOUR_KEY]` and `[YOUR_SECRET]` with your own keys from Minio.
```
DEBUG=electron-builder \
AWS_ACCESS_KEY_ID=[YOUR_KEY] \
AWS_SECRET_ACCESS_KEY=[YOUR_SECRET] \
yarn package \
--config.compression=store \
--config.publish.provider=s3 \
--config.publish.endpoint=http://127.0.0.1:9000 \
--config.publish.bucket=test-update \
--publish always
```

## Backup update feed files so that we can use later for testing update process
```
cd ~/minio-data/test-update/
cp beta-mac.yml beta-mac.yml-LATEST
```

## Downgrade the app version

```
vi package.json
vi app/package.json
```

## Rebuild the app
Replace `[YOUR_KEY]` and `[YOUR_SECRET]` with your own keys from Minio.
```
DEBUG=electron-builder \
AWS_ACCESS_KEY_ID=[YOUR_KEY] \
AWS_SECRET_ACCESS_KEY=[YOUR_SECRET] \
yarn package \
--config.compression=store \
--config.publish.provider=s3 \
--config.publish.endpoint=http://127.0.0.1:9000 \
--config.publish.bucket=test-update \
--publish always
```

## Replace update feed with newer version from first build
```
cd ~/minio-data/test-update/
cp beta-mac.yml-LATEST beta-mac.yml
```

## Run packaged app with logging enabled
```
DEBUG=zap* DEBUG_LEVEL=debug release/mac/ZapDesktop.app/Contents/MacOS/ZapDesktop
```

... It will check for updates when the app is started. It will also check every hour thereafter. If an update is found it will be downloaded and installed automatically and you should be presented with an option to either restart the app now or later.

screenshot 2018-07-27 12 43 48

This can be improved later by adding progress indicators, menu items to check for updates manually, a nicer ui etc.

### CODE REVIEW

File: app/updater.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        }, 100)
+      })
+    })
+
+    this.initAutoUpdate()
+  }
+
+  initAutoUpdate() {
+    autoUpdater.checkForUpdates()
+    const oneHour = 60 * 60 * 1000
```
Comment (JimmyMow) [inline app/updater.js]: is this the standard? curious how you came up with an hour

**Comment by JimmyMow** [comment]
utACK b5d70fa

File: app/updater.js (hunk 1 above)
Comment (mrfelton) [reply app/updater.js]: I just saw this in several examples and it seemed to make sense to me

**Comment by JimmyMow** [comment]
Tested ACK b5d70fa 🎊

**Pull request merged**